### PR TITLE
feat!: migrate to ESM

### PR DESCRIPTION
**Pull request by mdonnalley**
Migrate to ESM

@W-14490430@

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,43 +1,43 @@
 {
   "name": "plugin-template-sf-external",
   "description": "A template repository for external sf plugins",
-  "version": "0.0.1",
+  "version": "1.0.0",
   "dependencies": {
-    "@oclif/core": "^2.15.0",
+    "@oclif/core": "^3.11.0",
     "@salesforce/core": "^5.3.20",
```
Comment (mshanemc) [inline package.json]: change to core6

File: package.json [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-    "tslib": "^2"
+    "@salesforce/sf-plugins-core": "^5.0.1"
   },
   "devDependencies": {
-    "@salesforce/cli-plugins-testkit": "^4.4.7",
-    "@salesforce/dev-scripts": "^6.0.4",
-    "@swc/core": "^1.3.19",
+    "@salesforce/cli-plugins-testkit": "^5.0.4",
+    "@salesforce/dev-scripts": "^7.1.1",
     "@types/inquirer": "^8",
```
Comment (mshanemc) [inline package.json]: is types/inquirer necessary?

File: package.json [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
     "eslint-plugin-sf-plugin": "^1.16.15",
-    "oclif": "^3.17.2",
+    "oclif": "^4.0.3",
     "shx": "0.3.4",
     "ts-node": "^10.9.1",
-    "typescript": "^5.2.2"
+    "typescript": "^5.2.2",
+    "@oclif/plugin-command-snapshot": "^5.0.2",
+    "eslint-config-salesforce-typescript": "^3.0.5"
```
Comment (mshanemc) [inline package.json]: `eslint-config-salesforce-typescript` is not necessary because dev-scripts is bringing that.

File: package.json [hunk 4]
```diff
@@ -1,43 +1,43 @@
...
+    "@salesforce/dev-scripts": "^7.1.1",
     "@types/inquirer": "^8",
     "eslint-plugin-sf-plugin": "^1.16.15",
-    "oclif": "^3.17.2",
+    "oclif": "^4.0.3",
     "shx": "0.3.4",
     "ts-node": "^10.9.1",
-    "typescript": "^5.2.2"
+    "typescript": "^5.2.2",
+    "@oclif/plugin-command-snapshot": "^5.0.2",
```
Comment (mshanemc) [inline package.json]: nit: should be first in devDeps

File: package.json [hunk 5]
```diff
@@ -49,7 +49,8 @@
       "hello": {
         "description": "Commands to say hello."
       }
-    }
+    },
+    "flexibleTaxonomy": true
```
Comment (mshanemc) [inline package.json:51]: do you want a space separator?

File: package.json [hunk 6]
```diff
@@ -1,43 +1,43 @@
...
   "engines": {
-    "node": ">=16.0.0"
+    "node": ">=18.0.0"
   },
   "files": [
     "/lib",
     "/messages",
-    "/oclif.manifest.json"
+    "/oclif.manifest.json",
+    "/oclif.lock"
```
Comment (mshanemc) [inline package.json:27]: the "internal" equivalent has the script changes for doing oclif.lock, but this one doesn't.

Confirm that that's intentional.

File: package.json (hunk 5 above)
Comment (mdonnalley) [reply package.json:51]: already there on line 43

**Review by mshanemc — APPROVED** [review]

**Pull request merged**